### DISCUSSION TITLE
Policy (Usage control)

### DISCUSSION THREAD

**Discussion by Akshay-a-j** [eclipse-edc/Connector#2395] [General]
Is there a way where we can allow consumer to access part of the document (based on keys and values) instead of entire document?

**Comment by jimmarino** [comment]
What "document," what data, and when? Please be more specific about what you are requesting.

**Reply by Akshay-a-j** [reply]
Essentially, I want to share the test_document.txt file with company C, which contains information about companies A, B, and C. As a result, I want Company C to have access to only Company A's and Company C's data, but not Company B's data. For example:

test_document.txt contains following data:

#comapny A
{key1: value1 etc...}

#company B
{key2: value 2 etc...}

#company C
{key3: value 3}

So, Is there a way where we can allow company C to access part of the data (based on keys and values) instead of entire data?

**Comment by jimmarino** [comment]
The EDC doesn't make any requirements on the format of assets or how they are transferred. This is explained in the documentation. 

To achieve what you want, you must consider how to model your data appropriately. In the example you provided, it appears you have identified three assets, not one. Again, there is plenty of documentation that explains this, so please review it.

**Comment by Akshay-a-j** [comment]
Thank you @jimmarino for your quick responses. I believe I misphrased my question. I try to put it another way.
Assume I have a file called test document.txt that contains 20 lines of (xyz) data. Assume I share this test document.txt with two users (say, A and B), and I want user A to have access to ONLY the first ten lines of information (i.e., 1-10), and user B to have access to ONLY the last ten lines of information (i.e., 11-20). Is it possible to accomplish this? 
And can we do this using policy permissions\obligations\prohibitions?

**Comment by jimmarino** [comment]
What is a "user," e.g. a connector identity?  If so, please re-read my previous comment because it still applies. 

If "user" implies something else, then that concept needs to be handled outside of the EDC. The EDC has no concept (by design) of data-sharing identities other than the participant agent the connector represents. This is detailed extensively in the documentation.
